mealie/services/scraper/cleaner.py

```diff
@@ -95,15 +96,20 @@ def clean_image(image: str | list | dict | None = None, default="no image") -> s
     if not image:
         return default
 
-    match image:  # noqa - match statement not supported
+    match image:
         case str(image):
             return image
+        case {"url": str(url)}:
+            return url
         case list(image):
-            return image[0]
-        case {"url": str(image)}:
-            return image
-        case _:
-            raise TypeError(f"Unexpected type for image: {type(image)}, {image}")
+            for image_data in image:
+                match image_data:
+                    case str(image_data):
+                        return image_data
```

Comment:
I am not a 100% sure about this, but as far as i remember this will ultimately get used by the recipe_data_service to retrieve the image. That service however also accepts a list of image-urls and will use that to determine the largest one, so it would make sense here to not just return the first hit but the full list of urls so the best one can decided on further down the chain

Reply:
@fleshgolem I was also looking at this and confused on why this returns a string. We do have logic that fetches the largest image so I'm wondering if this should _really_ just return a list to begin with? 

@michael-genson I'm also not sure we need to nested match statement here - this is sort of what I was thinking

```python
match image:  # noqa - match statement not supported
    case str(image):
        return [image]
    case [str(_), *_]:
        return image
    case [{"url": str(_)}, *_]:
        return [x["url"] for x in image]
    case {"url": str(image)}:
        return [image]
    case _:
        raise TypeError(f"Unexpected type for image: {type(image)}, {image}")
```

That would catch all cases we're looking at now right? This returns the list, but could also just return the first resulting match as needed. I don't think we need recursion like I initially thought though.

Reply:
LGTM. I'll update it and make sure it passes the original issue's scrape